package.json

```diff
@@ -23,6 +23,7 @@
   },
   "devDependencies": {
     "@types/mocha": "^8.0.3",
+    "@types/node": "^14.14.31",
```

Comment:
We do need the Node.js types for our tests, but this is only a dev dependency, and we now have a test that `NodeJS.*` does not end up in `invariant.d.ts`, which is what matters here.

Reply:
I apologize if I've said this before, can't remember, I'll just say it anyway 🤷 
I just battled with this in an internal team project. Ended up going with `/// <reference types="jest" />` at the top of each individual test files. It was the lesser of 2 evils, but actually feels pretty good TBH. It lets us use the same `tsconfig.json` file for sources and tests, so no special setup there, and ensures VSCode doesn't complain about the test files not being part of the project. I also needed to add `types: []` to my tsconfig to ensure no source picks up `node_modules/@types/*` automatically. Only think still leaking is node type from my original linked issue, so after this change and that one last one like it, this setup will something something something complete.

Reply:
I should have mentioned one of the goals in that setup is to allow the popular pattern of colocating tests/specs with sources. No need for any of that if they are separate directories of course.